_posts/2020-09-15-open-sourcing-a-private-project.markdown

```diff
@@ -0,0 +1,87 @@
+
+Before we began the process of open sourcing we decided to make a list. We thought about what it would take to move the projects into the eyes of the public. We thought about what we wanted the app to look like, if there were any “secrets” in the code, and how we would present the projects. These are some of the things we came up with for our checklist.
+
+### Hiding sensitive information
+
+It is important to go through commit history and the code to make sure there are no “secrets” you would be sharing with the public when you open source. We ended up needing to squash the commit history because we had some areas that had sensitive information. Maybe your code is already free of sensitive information, but it is a good idea to check.
+
+Look at places where you may have hardcoded company specific or personal information. We often move these hardcoded pieces of code into environment variables. Just make sure you include the environment variables and an explanation in the README.
+
+### Creating a comprehensive README
+
+Having a good README is fundamental when creating an open source project. It is important for it to be robust and filled with easy to understand information on setting up and using the project. Some things that are important to include:
+
+- Description of the Project
+- Steps on how to setup the project
+- A contribution guide
+- Code of Conduct
+- Sponsorship (if applicable)
+- License
+
+You can read more about this over [here](https://www.fastruby.io/blog/open-source/ombulabs-open-source-guidelines.html) in our article about open source practices.
+
+### Moving the project to Open Source:
+
+The next step in open sourcing the project is to move it from Private to Public. We use [GitHub](https://github.com/) to store our repositories, so these steps are the ones for GitHub.
+To completely delete the history including Pull Requests it is necessary to create a fresh repository. If you didn’t have any sensitive information to hide, and there is no problem with sharing the commit history, you can simply go to settings on the repository page, enter the danger zone at the bottom of settings and click on `Change repository visibility`.
+
+If you do need to hide the history of the project there are several ways to do this, but this is the way we used:
+
+Note: before beginning this we made a copy on GitHub and locally in case anything went wrong.
+```
+$ git pull origin <main branch for repository >
+```
+Follow steps on [GitHub](https://docs.github.com/en/enterprise/2.17/user/github/administering-a-repository/deleting-a-repository) to delete the repository.
+
+Follow steps on [GitHub](https://docs.github.com/en/github/getting-started-with-github/create-a-repo) to create a new repository and make sure to choose the public option when choosing the visibility, and don’t include the README.
+```
+# Choose which commit you would like to include in the new tree(we chose the first commit. Use git log and copy the commit hash.
```

Comment:
I'd say that if you are looking to squash the last number of commits into a single one in the commit history, we can use the `git reset --soft [commit]` option. In our case we squashed all commits against the first one, but deleting the repo and recreating it would have given the same effect. There's an interesting side effect in this approach though, Github seems to detect contributions from other developers while if it was a repo from scratch it wouldn't.